### PR TITLE
Web bluetooth example

### PR DESCRIPTION
**Pull request by strangesast**
So there are examples of people using web-bluetooth with bleno/noble ([here's one](https://github.com/poshaughnessy/web-bluetooth-parrot-drone)), but I thought I'd make a slightly easier/accessible one with your pizza example.  The example I made is nearly identical to the noble code, but as it's a web page it also has a simple interface.
screenshot from 2016-09-08 12-07-01

### CODE REVIEW

**Comment by sandeepmistry** [comment]
Hi @strangesast,

This looks awesome!

What you do think about creating a new Github repo for it? Then we can add a link to it in [examples/pizza/README.md](https://github.com/sandeepmistry/bleno/blob/master/examples/pizza/README.md) similar to noble.

Let me know.

cc/ @monteslu @beaufortfrancois

File: examples/pizza/web/index.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+var cachedCharacteristics = {};
+
+// current bluetooth connection obj
+var ovenServer = null;
+
+// connect to bluetooth peripheral
+var readyOven = function() {
+  return navigator.bluetooth.requestDevice({
+    filters: [{services: serviceUuids}, {name: name}]
```
Comment (beaufortfrancois) [inline examples/pizza/web/index.js]: Nit: 
````js
filters: [{ services: serviceUuids, name: name }]

File: examples/pizza/web/index.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  if(ovenServer != null && ovenServer.connected) alert('Already connected...');
+  return readyOven().then(function() {
+    alert('Connection successful!');
+    console.log(ovenServer);
+  });
+};
+
+var onBakeButtonClick = function(e) {
+  if(ovenServer == null || !ovenServer.connected) alert('Not connected!');
+  [
```
Comment (beaufortfrancois) [inline examples/pizza/web/index.js]: The issue I'm seeing with this code is that `bakePizza` will be called even if `readyToppings` rejects sadly.
How about a simpler version?

```js
Promise.resolve()
.then(_ => readyCrust(getCrustType()))
.then(_ => readyToppings(getToppings()))
.then(_ => bakePizza(getOvenTemperature()))

**Comment by beaufortfrancois** [comment]
This is really really cool! I've should have thought doing this myself and I'm glad @strangesast did it ;)
I've left out a couple of nits that may be fixed before merge or creation of new repo.

@strangesast Why don't you host it in your own repo and @sandeepmistry will link to it and I will also link to it from https://github.com/WebBluetoothCG/demos if you don't mind.

File: examples/pizza/web/index.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    return pizzaCrustCharacteristic.writeValue(crust).catch(function(err) {
+      alert('crust error');
+      throw err;
+    });
+  }
+};
+
+var readyToppings = function(toppings) {
+  console.log('toppings:', toppings);
+  return function() {
```
Comment (beaufortfrancois) [inline examples/pizza/web/index.js]: QQ: Why do you `return function()` in these functions?

File: examples/pizza/web/index.html [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        return toppings;
+      };
+      
+      var getOvenTemperature = function() {
+        return ovenTempEl.value;
+      };
+      
+      // button listeners
+      var onStartButtonClick = function(e) {
+        if(ovenServer != null && ovenServer.connected) alert('Already connected...');
```
Comment (beaufortfrancois) [inline examples/pizza/web/index.html]: You may want to return after `alert`:

```js
        if(ovenServer != null && ovenServer.connected) {
          alert('Already connected...');
          return;
        }

Same applies for other alerts below.

File: examples/pizza/web/index.html [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        return readyOven().then(function() {
+          alert('Connection successful!');
+        });
+      };
+      
+      var onBakeButtonClick = function(e) {
+        if(ovenServer == null || !ovenServer.connected) alert('Not connected!');
```
Comment (beaufortfrancois) [inline examples/pizza/web/index.html]: You may want to return after `alert`:

```js
        if(ovenServer == null || !ovenServer.connected){
          alert('Not connected...');
          return;
        }

**Comment by beaufortfrancois** [comment]
One last nit!

**Comment by strangesast** [comment]
Thanks for the feedback!  I think I made it more complicated than necessary at first.  In addition to applying the suggestions from @beaufortfrancois, I moved everything to a single html file; I duplicated the functionality of peripheral.js in the others which is not what I wanted to demonstrate.

I created a new repo [here](https://github.com/strangesast/bleno-web-pizza-example).

File: examples/pizza/web/index.js (hunk 3 above)
Comment (strangesast) [inline examples/pizza/web/index.js]: When I had it in that `reduce` (which you pointed out earlier) I needed the closure.

**Comment by beaufortfrancois** [comment]
This is great!

One another suggestion I might add is to rename the `master` branch to `gh-pages` so that you could serve this from github.io.

**Comment by strangesast** [comment]
You can host gh pages from master (as of recently, maybe)
screenshot from 2016-09-13 09-01-18
So that's what I did ([example here](https://strangesast.github.io/bleno-web-pizza-example/
))

File: examples/pizza/web/index.html (hunk 5 above)
Comment (beaufortfrancois) [inline examples/pizza/web/index.html]: @strangesast (gentle ping)

**Comment by strangesast** [comment]
Thanks for the help, guys.  This project and web-bluetooth are awesome.  Glad to see that it's becoming more available (53 origin trial :D )

**Comment by beaufortfrancois** [comment]
I've submitted a PR to include your web app to pizza example README at https://github.com/sandeepmistry/bleno/pull/231

**Comment by sandeepmistry** [comment]
@strangesast new repo looks great!

One small suggestion, what do you think about splitting out the JS inside index.html to a separate file?

**Pull request closed without merging**